floor_annotation/lib/src/foreign_key.dart

```diff
@@ -41,7 +41,7 @@ abstract class ForeignKeyAction {
   /// When a parent key is modified or deleted from the database, no special
   /// action is taken. This means that SQLite will not make any effort to fix
   /// the constraint failure, instead, reject the change.
-  static const noAction = 1;
+  static const noAction = 0;
```

Comment:
💡 We could actually use an enum through all layers. In order to retrieve the enum values from `ForeignKey.onUpdate` and `ForeignKey.onDelete` annotation fields, the following function would do the trick. This would save us the conversion from `int` to `ForeignKeyAction` which would reduce complexity and future maintenance costs. What are your thoughts about this approach?

```dart
extension on DartObject {
  annotations.ForeignKeyAction toForeignKeyAction() {
    final enumValueString = toEnumValueString(); // defined in DartObjectExtension
    return annotations.ForeignKeyAction.values.singleWhere(
        (foreignKeyAction) => foreignKeyAction.toString() == enumValueString);
  }
}

// EntityProcessor._getForeignKeys
final onUpdateForeignKeyAction = foreignKeyObject
    .getField(ForeignKeyField.onUpdate)
    ?.toForeignKeyAction();
```

Reply:
Is fine for me!